contracts/test/integration/IntegrationTestVip13.t.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+import {ICore} from "../../core/ICore.sol";
+import {IVolt} from "../../volt/Volt.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {vip13} from "./vip/vip13.sol";
+import {TimelockSimulation} from "./utils/TimelockSimulation.sol";
+import {PriceBoundPSM} from "../../peg/PriceBoundPSM.sol";
+import {IPCVGuardian} from "../../pcv/IPCVGuardian.sol";
+
+contract IntegrationTestVIP13 is TimelockSimulation, vip13 {
+    uint256 public constant mintAmountDai = 1_000_000e18;
+    uint256 public constant mintAmountUsdc = 1_000_000e6;
+    uint256 public constant voltMintAmount = 1_000_000e18;
+
+    IERC20 public dai = IERC20(MainnetAddresses.DAI);
+    IERC20 public usdc = IERC20(MainnetAddresses.USDC);
+    ICore private core = ICore(MainnetAddresses.CORE);
+    IVolt oldVolt = IVolt(MainnetAddresses.VOLT);
+
+    function setUp() public {
+        mainnetSetup();
+        simulate(
+            getMainnetProposal(),
+            TimelockController(payable(MainnetAddresses.TIMELOCK_CONTROLLER)),
+            IPCVGuardian(MainnetAddresses.PCV_GUARDIAN),
+            MainnetAddresses.GOVERNOR,
+            MainnetAddresses.EOA_1,
+            vm,
+            false
+        );
+        mainnetValidate();
+
+        vm.startPrank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(this), mintAmountDai);
+        dai.transfer(address(voltV2DaiPriceBoundPSM), mintAmountDai * 2);
+        vm.stopPrank();
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        voltV2.mint(address(this), mintAmountDai);
+        core.revokeMinter(MainnetAddresses.GOVERNOR);
+        vm.stopPrank();
+
+        uint256 balance = usdc.balanceOf(MainnetAddresses.KRAKEN_USDC_WHALE);
+        vm.prank(MainnetAddresses.KRAKEN_USDC_WHALE);
+        usdc.transfer(address(this), balance);
+
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(MainnetAddresses.GOVERNOR);
+        voltV2.mint(address(voltV2UsdcPriceBoundPSM), voltMintAmount);
+        voltV2.mint(address(this), voltMintAmount);
+        vm.stopPrank();
+
+        usdc.transfer(address(voltV2UsdcPriceBoundPSM), balance / 2);
+    }
+
+    function testSwapDaiForVolt() public {
+        uint256 userStartingVoltBalance = voltV2.balanceOf(address(this));
+        uint256 minAmountOut = voltV2DaiPriceBoundPSM.getMintAmountOut(
```

Comment:
would be great to sanity check this number by getting the oracle price, taking mint `amountDai * 1e18 / oracle price` and then doing an assert approx equal on minAmountOut and the number we calculated